app/ota_client.py

```diff
@@ -503,28 +682,19 @@ def _update(
             url, cookies, metadata.get_persistent_info(), self._mount_point
         )
 
-        # finish update, we reset the downloader's proxy setting,
-        # although it is not needed actually
+        # standby slot preparation finished, set phase to POST_PROCESSING
+        logger.info("[update] update finished, entering post-update...")
+        self._update_phase = OtaClientUpdatePhase.POST_PROCESSING
+
+        # finish update, we reset the downloader's proxy setting
         self._download.cleanup_proxy()
 
-        # leave update
-        # wait for all subECUs before reboot itself
-        if post_update_event:
-            logger.info("waiting for all subECUs to become ready...")
-            if post_update_event.wait():
-                logger.info("all subECUs are ready")
-            else:
-                # upper caller timeout, failed to wait for all subECU to get ready
-                # raise an exception about this and report to the caller agent
-                #
-                # NOTE: this may cause the local ECU to be updated multiple
-                # times if any of a subECU keeps failing
-                msg = "cannot ensure all subECUs' are ready, abort current local ota update..."
-                logger.error(msg)
-                raise OtaErrorRecoverable(msg)
+        with fsm.proceed(fsm._P2, expect=fsm._S1) as _next:
+            assert _next == fsm._S2
+            logger.debug("[update] signal ota_service that local update finished")
 
-        logger.info("update finished, entering post-update...")
-        self._update_phase = OtaClientUpdatePhase.POST_PROCESSING
+        logger.debug("[update] leaving update, wait on ota_service and then reboot...")
```

Comment:
logger.info is prefer to save log.
```suggestion
        logger.info("[update] leaving update, wait on ota_service and then reboot...")
```